### PR TITLE
Add support for EKS

### PR DESCRIPTION
**Pull request by asutoshpalai**
1. **What this PR does / why we need it**:
This adds support for managing EKS clusters' lifecycle via TMC Terraform provider

2. **Which issue(s) this PR fixes**

        (optional, in `fixes #<issue number>` format, will close the issue(s) when PR gets merged):

        Fixes #

3. **Additional information**

4. **Special notes for your reviewer**

### CODE REVIEW

**Comment by Vasundharashukla** [comment]
@asutoshpalai we are planning to have acceptance tests as a part of this PR or separate?

File: internal/resources/ekscluster/flatten.go [hunk 1]
```diff
@@ -0,0 +1,301 @@
...
+*/
+
+package ekscluster
+
+import (
+	models "github.com/vmware/terraform-provider-tanzu-mission-control/internal/models/ekscluster"
+	"github.com/vmware/terraform-provider-tanzu-mission-control/internal/resources/common"
+)
+
+func flattenClusterSpec(item *models.VmwareTanzuManageV1alpha1EksclusterSpec) []interface{} {
```
Comment (Vasundharashukla) [inline internal/resources/ekscluster/flatten.go:13]: Can we have two resource files made, one for cluster and another for node pools and add the construct and flatten functions respectively in each of them. That will make it in sync with how other resources have been implemented.

File: internal/resources/ekscluster/flatten_test.go [hunk 2]
```diff
@@ -0,0 +1,1460 @@
...
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/require"
+
+	eksmodel "github.com/vmware/terraform-provider-tanzu-mission-control/internal/models/ekscluster"
+)
+
+func TestFlattenCluterSpec(t *testing.T) {
```
Comment (Vasundharashukla) [inline internal/resources/ekscluster/flatten_test.go:16]: Same goes for the flatten tests, dividing the eks resource into two and taking the tests separately will make it more organised. Also, let's avoid having more than 1000 lines of code in one file.

**Review by Vasundharashukla — CHANGES_REQUESTED** [review]

**Comment by asutoshpalai** [comment]
Acceptance tests will be a separate PR, which will be merged into this branch.

File: internal/resources/ekscluster/flatten_test.go (hunk 2 above)
Comment (asutoshpalai) [reply internal/resources/ekscluster/flatten_test.go]: done

File: internal/resources/ekscluster/flatten.go (hunk 1 above)
Comment (asutoshpalai) [reply internal/resources/ekscluster/flatten.go]: done

File: internal/authctx/helper.go [hunk 3]
```diff
@@ -94,6 +95,32 @@ func ProviderConfigureContext(_ context.Context, d *schema.ResourceData) (interf
 	return setContext(&config)
 }
 
+func ProviderConfigureContextWithDefaultTransport(_ context.Context, d *schema.ResourceData) (interface{}, diag.Diagnostics) {
```
Comment (shreyassreenivas) [inline internal/authctx/helper.go:98]: The comments added is not reflecting here> 
and this function name should be renamed to reflect that it is used for Tests only

**Review by shreyassreenivas — DISMISSED** [review]

File: internal/authctx/client.go [hunk 4]
```diff
@@ -36,9 +36,18 @@ type TanzuContext struct {
...
 	cfg.TMCConnection, err = client.NewHTTPClient(cfg.TLSConfig)
 	if err != nil {
-		return
+		return err
 	}
 
+	return setup(cfg)
+}
+
+func (cfg *TanzuContext) SetupWithDefaultTransport() (err error) {
```
Comment (shreyassreenivas) [inline internal/authctx/client.go:45]: Then you need to change this function to Test as well

**Review by Vasundharashukla — APPROVED** [review]
LGTM

**Review by shreyassreenivas — APPROVED** [review]

**Pull request merged**